shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java

```diff
@@ -139,15 +138,8 @@ private void doHealthCheck() {
     }
 
     private void check(final Map<String, List<DivideUpstream>> map) {
-        for (Map.Entry<String, List<DivideUpstream>> entry : map.entrySet()) {
-            String key = entry.getKey();
-            List<DivideUpstream> value = entry.getValue();
-            for (DivideUpstream upstream : value) {
-                CompletableFuture<UpstreamWithSelectorId> future = CompletableFuture.supplyAsync(() -> check(key, upstream), executor);
-
-                futures.add(future);
-            }
-        }
+        map.forEach((k, v) -> v.forEach(i -> futures.add(
+                CompletableFuture.supplyAsync(() -> check(k, i), executor).thenAccept(d -> putEntityToMap(d)))));
```

Comment:
`.thenAccept(d -> putEntityToMap(d)` That's not the original logic.

Reply:
The original logic is done in waitFinish.   That's not the correct way to use CompletableFuture.

Reply:
The original way to do putEntityToMap is in waitFinish, it's after `check` right ?

Reply:
Can you help to remove `waitFinish` method and use the correct way?

Reply:
Why remove `waitFinish` ? it's waiting for the futures to done.

Reply:
My mistake. You already removed ` putEntityToMap(entity); `from `waitFinish `